cmd/ncproxy/ncproxy.go

```diff
@@ -557,6 +558,58 @@ func (s *grpcService) GetNetworks(ctx context.Context, req *ncproxygrpc.GetNetwo
 // client connections.
 type ttrpcService struct {
 	m sync.Mutex
+	// database for containerID to compute agent address
+	agentStore *computeAgentStore
+}
+
+func newTTRPCService(ctx context.Context, db *bolt.DB) *ttrpcService {
+	agentStore := newComputeAgentStore(db)
+	return &ttrpcService{
+		agentStore: agentStore,
+	}
+}
+
+func getComputeAgentClient(agentAddr string) (computeagent.ComputeAgentService, error) {
+	conn, err := winio.DialPipe(agentAddr, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to compute agent service")
+	}
+	client := ttrpc.NewClient(
+		conn,
+		ttrpc.WithUnaryClientInterceptor(octtrpc.ClientInterceptor()),
+		ttrpc.WithOnClose(func() { conn.Close() }),
+	)
+	return computeagent.NewComputeAgentClient(client), nil
+}
+
+func (s *ttrpcService) reconnectComputeAgents(ctx context.Context) {
+	contents, err := s.agentStore.getActiveComputeAgents(ctx)
+	if err != nil && errors.Is(err, errBucketNotFound) {
+		// no entries in the database yet, return early
+		log.G(ctx).WithError(err).Debug("no entries in database")
+		return
+	} else if err != nil {
+		log.G(ctx).WithError(err).Warn("failed to get compute agent information")
```

Comment:
Are there expected cases where we would hit this? If not I would upgrade the log level to `Error`.